src/Writer/PhpArray.php

```diff
@@ -139,4 +158,54 @@ protected function processIndented(array $config, array $arraySyntax, &$indentLe
 
         return $arrayString;
     }
+
+    /**
+     * Process a string configuration value
+     *
+     * @param string $value
+     * @return string
+     */
+    protected function processStringValue($value)
+    {
+        if ($this->useClassNameScalars && $this->checkStringIsFqn($value)) {
+            return $value . '::class';
+        }
+
+        return var_export($value, true);
+    }
+
+    /**
+     * Process a string configuration key
+     *
+     * @param string $key
+     * @return string
+     */
+    protected function processStringKey($key)
+    {
+        if ($this->useClassNameScalars && $this->checkStringIsFqn($key)) {
+            return $key . '::class';
+        }
+
+        return "'" . addslashes($key) . "'";
+    }
+
+    /**
+     * Check whether a string represents a resolvable FQCN
+     *
+     * @param string $string
+     * @return bool
+     */
+    protected function checkStringIsFqn($string)
```

Comment:
Otherwise use `explode` and match each segment, but the point is that you need to match the regex in order to prevent errors in cases where `class_alias()` was used inside any codebase (because [this stuff can happen](https://3v4l.org/ApMIA))